src/main/resources/org/jenkinsci/plugins/docker/workflow/Docker.groovy

```diff
@@ -80,21 +80,36 @@ class Docker implements Serializable {
             def parsedArgs = args.split(/ (?=([^"']*["'][^"']*["'])*[^"']*$)/)
             def dir = parsedArgs[-1] ?: '.'
 
-            // Detect custom Dockerfile:
+            // iterate through arguments to check for extra variables that need to be set
+            def target = null
             def dockerfile = "${dir}/Dockerfile"
             for (int i=0; i<parsedArgs.length; i++) {
                 def arg = parsedArgs[i]
-                if ((arg == '-f' || arg.startsWith('--file')) && i < (parsedArgs.length - 1)) {
-                    dockerfile = arg.startsWith('--file=') ? arg.split('=')[1] : parsedArgs[i+1]
-                    break
+
+                // determine build target
+                if (arg.startsWith('--target=')) {
+                    target = arg.split('=')[1]
+                }
+                else if (arg == '--target' && i < (parsedArgs.length - 1)) {
+                    target = parsedArgs[i+1]
```

Comment:
These two lines could be collapsed if you replaced `i+1` with `i++` -- but i'm not sure how comfortable people are with doing that...